contracts/libraries/OracleLibrary.sol

```diff
@@ -76,4 +76,34 @@ library OracleLibrary {
 
         return uint32(block.timestamp) - observationTimestamp;
     }
+
+    /// @notice Given a pool, it returns the tick value as of the start of the current block
+    /// @param pool Address of Uniswap V3 pool
+    /// @return The tick that the pool was in at the start of the current block
+    function getBlockStartingTick(address pool) internal view returns (int24) {
+        (, int24 tick, uint16 observationIndex, uint16 observationCardinality, , , ) = IUniswapV3Pool(pool).slot0();
+
+        // 3 observations are needed to reliably calculate the block starting tick
+        require(observationCardinality > 2, 'NEO');
+
+        // If the latest observation occurred in the past, then no tick-changing trades have happened in this block
+        // therefore the tick in `slot0` is the same as at the beginning of the current block.
+        // We don't need to check if this observation is initialized - it is guaranteed to be.
+        (uint32 observationTimestamp, , , ) = IUniswapV3Pool(pool).observations(observationIndex);
+        if (observationTimestamp != block.timestamp) {
+            return tick;
+        }
+
+        uint256 prevIndex = (observationIndex + observationCardinality - 1) % observationCardinality;
+        int56 tickCumulative;
+        bool initialized;
+        (observationTimestamp, tickCumulative, , initialized) = IUniswapV3Pool(pool).observations(prevIndex);
```

Comment:
right, so this is just off by 1, hence my comment earlier about the required # of observations being 2, not 3.

so actually, on L92 you should _also_ fetch the `tickCumulative` from `observations(observationIndex)` (we don't need to worry about this observation being initialized, it's guaranteed once we add the `observationCardinality > 0` check above)

then, here, we should use `prevIndex` (which looks right to me) to fetch the prior observation , make sure it's initialized, and perform the delta calculation just like you did